src/workflows/server/server.py

```diff
@@ -855,6 +864,18 @@ async def _extract_run_params(self, request: Request, workflow: Workflow) -> tup
             context = None
             if context_data:
                 context = Context.from_dict(workflow=workflow, data=context_data)
+            elif handler_id:
+                persisted_handlers = await self._workflow_store.query(
+                    HandlerQuery(
+                        handler_id_in=[handler_id],
+                        workflow_name_in=[workflow_name],
+                        status_in=["completed"],
+                    )
+                )
+                if len(persisted_handlers) == 0:
+                    raise HTTPException(detail="Handler not found", status_code=404)
+
+                context = Context.from_dict(workflow, persisted_handlers[0].ctx)
```

Comment:
just confirming, 'cause I don't know, but the handler id is specifiable via the context, right? Or will this make a new handler id?